backend/bot/index.js

```diff
@@ -101,22 +90,42 @@ const postRankingChangeMessage = async (oldRankings, newRankings) => {
       `${c.name} ${c.oldRanking}. ⟶ ${c.newRanking}.`)
     )
     .join('\n')
-  await bot.postMessage(channel.id, messageText, { as_user: true });
+  return messageText
 }
 
 const hasLeaderboardChanged = (leaderboardSize, oldRankings, newRankings) => (
   oldRankings.findIndex((oldPlayer, index) => oldPlayer.id !== newRankings[index].id) < leaderboardSize
 )
 
-const updatePurpose = async (rankings) => {
+const createPurposeMessage = async (rankings) => {
   const rankingsText = rankings
     .map((ranking, i) => `${i + 1}. ${ranking.name} (${ranking.rating})`)
     .join('\n')
   const purpose = 'TOP PLAYERS\n' + rankingsText
-  await bot.setGroupPurpose(channel.id, purpose)
+  return purpose
 }
 
 
+const makeBot = async (botToken, channelName) => {
+  return new Promise((resolve, reject) => {
+    if (!botToken || !channelName) {
+      return reject(Error(`Missing bot token or channel name, bot didn't start!`))
+    }
+
+    const slackBot = new SlackBot({
```

Comment:
i really really dislike this part, where you asynchronously initialize the SingleChannelBot.
is this a normal pattern in javascript?
ideally if I could initialize the slackbot and groupId inside the singlechannelbot, but how? 🤷‍♂

Reply:
Not sure if this is what you want and if it would even work but maybe you can try to make the constructor in `SingleChannelBot` async and then initialize everything there?
https://stackoverflow.com/a/50885340/1274911

Reply:
ok, i'll keep in like it is for now